sqldelight-compiler/dialect/src/main/kotlin/app/cash/sqldelight/dialect/api/SqlDelightDialect.kt

```diff
@@ -9,16 +9,26 @@ interface SqlDelightDialect {
    */
   val driverType: ClassName
 
+  /**
+   * Dialect-specific implementation of an `AsyncSqlDriver`.
+   * TODO: Check if this needs to be nullable, also maybe combine these into a class
```

Comment:
Had the same thought about sticking them in a class. `RuntimeTypes` or something like that. I think its a good idea

Reply:
Do you want that in this PR? I guess may as well get it out of the way while adding the async types?

Reply:
yea if it can be done in this PR I'd just do it now, I think it'll shrink some bits of it